### PR TITLE
Feature/move file operations to model

### PR DESCRIPTION
**Pull request by jamescdavis**
## Ticket

?

# Purpose

Move file operations to file model/adapter.

# Notes for Reviewers

Make sure code is sane.

### CODE REVIEW

File: addon/components/file-browser/component.js [hunk 1]
```diff
@@ -294,9 +272,10 @@ export default Ember.Component.extend({
                         this.get('_items').removeObject(replacedItem);
                         this.notifyPropertyChange('_items');
                     }, 1800);
+                    // Later to avoid flash() trying to set on a destroyed item.
+                    Ember.run.later(() => replacedItem.unloadRecord(), 2200);
```
Comment (jamescdavis) [inline addon/components/file-browser/component.js:276]: I'm not really happy with this, but...

Comment (jamescdavis) [reply addon/components/file-browser/component.js:276]: Alternative would be to allow flash() to take a callback?

Comment (hmoco) [reply addon/components/file-browser/component.js:276]: We could do that. I also never had problems with flash being called in a destroyed item, which I guess I should've.

**Comment by hmoco** [comment]
Code looks good, gonna do a functional review

**Pull request merged**